libgit/repo.go

```diff
@@ -262,3 +266,52 @@ func CreateRepoAndID(
 	}
 	return id, err
 }
+
+// DeleteRepo "deletes" the given repo in the given TLF.  RIght now it
+// simply moves the repo out of the way to a special directory, to
+// allow any concurrent writers to finish their pushes without
+// triggering conflict resolution.  The caller is responsible for
+// syncing the FS and flushing the journal, if desired.  It expects
+// the `config` object to be unique during the lifetime of this call.
+func DeleteRepo(
+	ctx context.Context, config libkbfs.Config, tlfHandle *libkbfs.TlfHandle,
+	repoName string) error {
+	// Create a unique ID using the verifying key and the `config`
+	// object, which should be unique to each call in practice.
+	session, err := config.KBPKI().GetCurrentSession(ctx)
+	if err != nil {
+		return err
+	}
+	uniqID := fmt.Sprintf("%s-%p", session.VerifyingKey.String(), config)
+
+	kbfsOps := config.KBFSOps()
+	rootNode, _, err := kbfsOps.GetOrCreateRootNode(
+		ctx, tlfHandle, libkbfs.MasterBranch)
+	if err != nil {
+		return err
+	}
+	normalizedRepoName := normalizeRepoName(repoName)
+
+	repoNode, _, err := kbfsOps.Lookup(ctx, rootNode, kbfsRepoDir)
+	if err != nil {
+		return err
+	}
+
+	_, _, err = kbfsOps.Lookup(ctx, repoNode, normalizedRepoName)
+	if err != nil {
+		return err
+	}
+
+	deletedReposNode, err := lookupOrCreateDir(
+		ctx, config, repoNode, kbfsDeletedReposDir)
+	if err != nil {
+		return err
+	}
+
+	// For now, just rename the repo out of the way, using the uniq
+	// ID.  TODO(KBFS-2442): periodically delete old-enough repos from
+	// `kbfsDeletedReposDir`.
```

Comment:
So `kbfsDeletedReposDir` is a staging area for deleted repos? Meaning anything there is good to be deleted?

Reply:
Yes, though it's probably best to wait until the renamed repo has been deleted at least a few minutes, to make sure any concurrent writers have finished pushing to it.